src/lib/containers/markdown/MarkdownEditor.tsx

```diff
@@ -17,7 +17,85 @@ export const MarkdownEditor: React.FunctionComponent<MarkdownEditorProps> = ({
   const [currentTab, setCurrentTab] = useState<MarkdownEditorTabs>(
     MarkdownEditorTabs.WRITE,
   )
-  const [text, setText] = useState<string>()
+  const [text, setText] = useState<string>('')
+  const [selectionStart, setSelectionStart] = useState<number>(0)
+  const commandList = [
+    'title',
+    'bold',
+    'italic',
+    'strikethrough',
+    'code',
+    'latex',
+    'subscript',
+    'superscript',
+    'link',
+    'image',
+  ]
+
+  const textAreaRef = useRef<HTMLTextAreaElement>(null)
+
+  useEffect(() => {
+    const input = textAreaRef.current
+    if (input) {
+      input.setSelectionRange(selectionStart, selectionStart)
+    }
+  }, [textAreaRef, selectionStart])
+
+  const handleCommands = (command: string) => {
+    const textVal = textAreaRef.current
+    if (textVal) {
+      const start = textVal.selectionStart
+      const end = textVal.selectionEnd
+      const selected = text.substring(start, end)
+      const first = text.substring(0, start)
+      const later = text.substring(end, text.length)
+
+      let openSyntax = ''
+      let closeSyntax = ''
+      if (command === 'bold') {
+        openSyntax = closeSyntax = '**'
```

Comment:
I think it might be cleaner to just create a new file with a data object like
```ts
type MarkdownEditorCommandData = { // or a better name if you can come up with one
	openSyntax: string
	closeSyntax: string | null
}

type MarkdownEditorCommands = Record<CommandListType, MarkdownEditorCommandData>

export const MARKDOWN_COMMANDS_DATA: MarkdownEditorCommands: {
	bold: {
		openSyntax: "**",
		closeSyntax: "**"
	},
	title: {
		openSyntax: "###",
		closeSyntax: null
	},
	// ...
}
```

Then just pull it in here and use it.